### PR TITLE
New enter weights form

### PR DESCRIPTION
**Pull request by csebranek**
-Includes the new weights form to replace the old ext3 form, with a table top restraint field built in as well
-Updates to manually apply in LabKey after this code change are: add restraint_objectid as a text field to the weights dataset and also set the permissions in the restraints dataset to match the ones in the weights dataset, update ehr.form_framework_types to point to the weight form

### CODE REVIEW

File: WNPRC_EHR/src/feeding/base/FeedingForm.tsx [hunk 1]
```diff
@@ -118,7 +118,6 @@ const FeedingForm: React.FunctionComponent<any> = (props) => {
             onChange={handleDateChange}
             customInput={
               <DateInput
-                value={values.date.value}
```
Comment (dnicolalde) [inline WNPRC_EHR/src/feeding/base/FeedingForm.tsx:121]: Did you remove this by choice?

File: WNPRC_EHR/src/typings/main.ts [hunk 2]
```diff
@@ -5,4 +5,37 @@ export interface ConfigProps {
     sort?: string;
     containerPath?: string;
     filterArray?: Array<any>;
-}
\ No newline at end of file
+}
```
Comment (dnicolalde) [inline WNPRC_EHR/src/typings/main.ts:8]: Is this the animal abstract for the weight form?

File: WNPRC_EHR/src/weight/components/AnimalInfoPane.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+  if (infoState == "loading") {
+    return (
+      <div id="animal-info-empty">
+        <Spinner text={"Loading..."} />
+      </div>
+    );
+  }
+
+  if (infoState == "loading-success") {
```
Comment (dnicolalde) [inline WNPRC_EHR/src/weight/components/AnimalInfoPane.tsx:29]: I though we move away from hard coding html in the type script files?

File: WNPRC_EHR/src/weight/typings/main.d.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+/* App globals */
```
Comment (dnicolalde) [inline WNPRC_EHR/src/weight/typings/main.d.ts:1]: Is this file needed?

**Review by dnicolalde — COMMENTED** [review]
Chad can you look at the comments I added and make sure I am understanding at least the animal info panel. Is this different than the one you are using on the animal history?

File: WNPRC_EHR/src/feeding/base/FeedingForm.tsx (hunk 1 above)
Comment (csebranek) [reply WNPRC_EHR/src/feeding/base/FeedingForm.tsx:121]: yes, it's not needed and it does not affect the behavior of the date input.

File: WNPRC_EHR/src/typings/main.ts (hunk 2 above)
Comment (csebranek) [reply WNPRC_EHR/src/typings/main.ts:8]: The AnimalInfoPane.tsx is the animal abstract for the weight form yes. It is different from the one used in the animal history.

File: WNPRC_EHR/src/weight/components/AnimalInfoPane.tsx (hunk 3 above)
Comment (csebranek) [reply WNPRC_EHR/src/weight/components/AnimalInfoPane.tsx:29]: Not sure what you mean here, this basically determines what html should render across different "animal info" states, e.g., animal not found, animal found, etc.

File: WNPRC_EHR/src/weight/typings/main.d.ts (hunk 4 above)
Comment (csebranek) [reply WNPRC_EHR/src/weight/typings/main.d.ts:1]: Nope, this is probably not needed, it's a relic of some initial config :)

**Comment by csebranek** [comment]
Hi @dnicolalde , please see my responses above and resolve if you have enough info.

File: WNPRC_EHR/src/weight/components/AnimalInfoPane.tsx (hunk 3 above)
Comment (csebranek) [reply WNPRC_EHR/src/weight/components/AnimalInfoPane.tsx:29]: Ah I think I know what you mean, referring to the part below. Yes, we did do this change for the animal history version but not for this, I can add this as a TODO.

Comment (dnicolalde) [reply WNPRC_EHR/src/weight/components/AnimalInfoPane.tsx:29]: Yeah that will be good, I remember you mentioned that we went away from hard coding and using a view to custom the animal abstract
